components/client/src/main/java/com/hotels/styx/client/netty/NettyToStyxResponsePropagator.java

```diff
@@ -104,7 +105,15 @@
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
         ensureContentProducerIsCreated(ctx);
-        contentProducer.ifPresent(producer -> producer.channelException(new BadHttpResponseException(origin, cause)));
+        contentProducer.ifPresent(producer -> producer.channelException(toStyxException(cause)));
+    }
+
+    private RuntimeException toStyxException(Throwable cause) {
+        if (cause instanceof OutOfMemoryError) {
+            return new StyxClientError("Styx Client: Not enough memory.", cause);
```

Comment:
Perhaps we can make the stack trace more readable by specifying which kind of memory we have exhausted, since this condition will cover all kinds. The easiest way is probably to append the message of the parent error.